api/src/pcapi/core/subscription/api.py

```diff
@@ -93,13 +93,16 @@ def has_completed_profile(user: users_models.User, eligibility: users_models.Eli
     ).scalar()
 
 
-def _has_completed_profile_in_dms_form(user: users_models.User) -> bool:
+def _has_completed_profile_in_dms_form(
+    user: users_models.User, eligibility: users_models.EligibilityType | None
```

Comment:
La propriété `eligibility` est-elle associée à un utilisateur ?

En fait, je pense que je perçois que tu as besoin d'ajouter ce paramètre pour gérer ton cas particulier, et pourtant je trouve curieux qu'on ait besoin d'ajouter un paramètre `eligibility` qui semble intrinsèquement lié à `user` que tu donnes déjà à la fonction.

Mais je me trompe peut-être sur le modèle user.

Reply:
Tu peux pas utiliser la fonction `_get_activable_identity_check` au lieu de passer `eligibility` en cascade ? pareil pour should_complete_profile

Reply:
J'ai renommé `should_complete_profile` en `not has_completed_profile_for_given_eligibility`